uvicorn/protocols/websockets/wsproto_impl.py

```diff
@@ -232,17 +232,14 @@ def send_500_response(self) -> None:
             (b"content-type", b"text/plain; charset=utf-8"),
             (b"connection", b"close"),
         ]
-        if self.conn.connection is None:
-            output = self.conn.send(wsproto.events.RejectConnection(status_code=500))
-        else:
-            msg = h11.Response(
-                status_code=500, headers=headers, reason="Internal Server Error"
+        output = self.conn.send(
+            wsproto.events.RejectConnection(
+                status_code=500, headers=headers, has_body=True
             )
-            output = self.conn.send(msg)
-            msg = h11.Data(data=b"Internal Server Error")
-            output += self.conn.send(msg)
-            msg = h11.EndOfMessage()
-            output += self.conn.send(msg)
+        )
+        output += self.conn.send(
+            wsproto.events.RejectData(data=b"Internal Server Error")
+        )
```

Comment:
Don't be afraid here. I'll explain what's happening.

1. We were sending `RejectConnection(status_code=500)` without a body, but on `websockets` implementation we were sending the "Internal Server Error" body on the analogous behavior. The `RejectData` matches the behavior.
2. We are removing the conditional because it's never reached, and the reason for it is that we only call `send_500_response` is we didn't complete the handshake - which makes a lot of sense to remove it.

Hope it's clear.

Reply:
Am I missing `EndOfMessage`? I need to check this.

Reply:
No, I'm not missing it. `wsproto` internally adds the `h11.EndOfMessage`. :pray: